plugins/anomalydetection/src/plugin.cpp

```diff
@@ -0,0 +1,817 @@
+            break;
+        // todo better unit tests and double check the parent lineage traversal fields in general
+        case plugin_sinsp_filterchecks::TYPE_SNAME:
+        {
+            int64_t sid;
+            m_sid.read_value(tr, thread_entry, sid);
+            m_ptid.read_value(tr, thread_entry, ptid);
+            falcosecurity::table_entry last_entry(nullptr, nullptr, nullptr);
+            falcosecurity::table_entry* leader = &thread_entry;
+            for(uint32_t j = 0; j < 9; j++)
+            {
+                try
+                {
+                    auto lineage = m_thread_table.get_entry(tr, ptid);
+                    m_sid.read_value(tr, lineage, tint64);
+                    if(sid != tint64)
+                    {
+                        break;
+                    }
+                    m_ptid.read_value(tr, lineage, ptid);
+                    last_entry = std::move(lineage);
+                    leader = &last_entry;
+                }
+                catch(const std::exception& e)
+                {
+                }
+            }
+            m_comm.read_value(tr, *leader, tstr);
+            break;
+        }
+        case plugin_sinsp_filterchecks::TYPE_VPGID:
+            m_vpgid.read_value(tr, thread_entry, tint64);
+            tstr = std::to_string(tint64);
+            break;
+        case plugin_sinsp_filterchecks::TYPE_VPGID_NAME:
+        {
+            int64_t vpgid;
+            m_vpgid.read_value(tr, thread_entry, vpgid);
+            m_ptid.read_value(tr, thread_entry, ptid);
+            falcosecurity::table_entry last_entry(nullptr, nullptr, nullptr);
+            falcosecurity::table_entry* leader = &thread_entry;
+            for(uint32_t j = 0; j < 5; j++)
+            {
+                try
+                {
+                    auto lineage = m_thread_table.get_entry(tr, ptid);
+                    m_vpgid.read_value(tr, lineage, tint64);
+                    if(vpgid != tint64)
+                    {
+                        break;
+                    }
+                    m_ptid.read_value(tr, lineage, ptid);
+                    last_entry = std::move(lineage);
+                    leader = &last_entry;
+                }
+                catch(const std::exception& e)
+                {
+                }
+            }
+            m_comm.read_value(tr, *leader, tstr);
+            break;
+        }
+
+        //
+        // fd or fs related
+        //
+
+        case plugin_sinsp_filterchecks::TYPE_FDNAME:
+        {
+            switch(evt.get_type())
+            {
+            case PPME_SYSCALL_OPEN_X:
+            case PPME_SOCKET_ACCEPT_X:
+            case PPME_SOCKET_ACCEPT_5_X:
+            case PPME_SOCKET_ACCEPT4_X:
+            case PPME_SOCKET_ACCEPT4_5_X:
+            case PPME_SOCKET_ACCEPT4_6_X:
+            case PPME_SYSCALL_CREAT_X:
+            {
+                // todo get fd and look up the right fd from the table
```

Comment:
@jasondellaluce How do you get the current fd and then do a targeted lookup in the fd table. libs has m_lastevent_fd ... Thanks in advance!

Reply:
We briefly chatted on slack and `lastevent_fd` is indeed not accessible, hence had to write it to the thread_entry within the plugin. It probably makes sense to expand the plugin API a bit further and make those fields accessible as well so that the work does not have to be done twice.